src/EditorFeatures/CSharpTest/ConvertForToForEach/ConvertForToForEachTests.cs

```diff
@@ -883,6 +883,193 @@ void Test(string[] array)
             Console.WriteLine(array[i]);
         }
     }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertForToForEach)]
+        public async Task TestMultidimensionalArray1()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System;
+
+class C
+{
+    void Test(string[,] array)
+    {
+        [||]for (int i = 0; i < array.Length; i++)
+        {
+            Console.WriteLine(array[i, 0]);
```

Comment:
It would be interesting to see what happens if you walk along the diagonal with array[i ,i]

Reply:
#resolved  We do the right thing (not offer the refactoring)